decisions/0002-courseware-migration-plan.rst

```diff
@@ -0,0 +1,176 @@
+provide us. The second is to focus exclusively on replicating existing
+functionality first, so that the transition can happen as quickly as possible,
+even if signficant pieces of that work will get thrown away later.
+
+The approach outlined here takes a middle road, with the goal of building a more
+powerful Studio authoring experience while minimizing the necessary upfront
+changes to the LMS. Some high level principles:
+
+1. Everything currently authored in Modulestore or Contentstore will be moved to
+   Blockstore eventually.
+2. The transition must be gradual and we will be running in a mixed environment
+   for some time.
+3. Developing powerful and flexible authoring is a higher priority than
+   improving LMS storage.
+
+
+------------
+Target State
+------------
+
+Once this plan is executed, the proposed end state is:
+
+1. Everything from Modulestore and Contentstore is moved to Blockstore on the
+   authoring side.
+2. The LMS continues to use Modulestore to serve courseware (and other things).
+3. The LMS points to static assets controlled by Blockstore.
+4. Content Libraries will become a more general concept, and the Content Library
+   of today will be recast as a type of Content Library, like "Problem Bank".
+   With the idea that additional, pluggable types will follow.
+5. The publish step explicitly pulls data from Blockstore into the Split
+   Modulestore.
+
+This would get us to a place where we could lay the foundation for a more
+powerful (search/tagging/reuse) authoring experience, without having to boil the
+ocean to rework the existing Modulestore for serving courseware to students.
+It would also allow us to separately evolve the LMS's read-optimized data store
+from Blockstore's read/write use cases, as well as any other read-optimized
+stores and applications as the system evolves to more dynamic use cases.
+
+Note that this is the target state where we can declare that all content
+authoring has been migrated to Blockstore. This is not the end state of the
+system as a whole, and the number of places Blockstore publishes to would likely
+increase to support things like adaptive learning use cases.
+
+
+----------------
+Release Timeline
+----------------
+
+This is a high level summary of the release timeline, with more details about
+each piece in the sections that follow. The goal is to always give a transition
+release cycle for backwards compatibility, and to provide new Blockstore-powered
+capabilities with each release.
+
+Juniper
+=======
+
+* Old Mongo: Management command to convert Old Mongo courses to Split.
+* XBlock Conversion: Video, HTML, and Capa XModules are converted to XBlock.
+* Content Libraries: Mixed operation where existing Content Libraries are
+  supported with a conversion management command. Converted Content Libraries
+  (Problem Banks) will have a new authoring experience making them easier to
+  search and manage.
```

Comment:
Will this use the existing XBlock/architecture where the xblocks in the Library are copied over into the course?

Reply:
Yes. The way content libraries currently copy content is ugly, but it makes for a really easy interface to swap an implementation behind.